src/TaskModel.vala

```diff
@@ -559,6 +585,10 @@ public class Tasks.TaskModel : Object {
 
             client.generate_instances_for_object_sync (comp, start.as_timet (), end.as_timet (), null, recur_instance_callback);
         }
+
+        if (ECal.util_component_has_x_property (comp, "X-EVOLUTION-GTASKS-POSITION")) {
+            yield update_gtasks_tasklist (list, client, get_registry_sync ());
+        }
```

Comment:
Why do we need to trigger a refresh here? Once a task is completed, isn't it enough to just hide it from the UI?